agent/api/task/task.go

```diff
@@ -1882,6 +2024,37 @@ func (task *Task) shouldOverrideNetworkModeAwsvpc(container *apicontainer.Contai
 	return true, dockerMappingContainerPrefix + pauseContainer.DockerID
 }
 
+// shouldOverrideNetworkModeServiceConnectBridge checks if a bridge-mode SC task container needs network mode override
+// For non-internal containers in an SC bridge-mode task, each gets a pause container, and should be launched
+// with container network mode use pause container netns (the "docker run" equivalent option is
+// "--network container:$pause_container_id")
+func (task *Task) shouldOverrideNetworkModeServiceConnectBridge(container *apicontainer.Container, dockerContainerMap map[string]*apicontainer.DockerContainer) (bool, string) {
+	if container == task.GetServiceConnectContainer() {
+		return false, "" // SC container will use bridge mode (it doesn't need a pause container)
+	}
+	pauseContainer, err := task.getServiceConnectBridgeModePauseContainerForTaskContainer(container)
```

Comment:
@rajal-amzn I used a map in the first commit but later changed it because Angel brought up a good point that if we keep a map then this map will need to configure it be to stored in Agent database if Agent restarts, so we thought if we could simply use the naming convention for resolving either pause or task container with the other it'll be more convenient.

Reply:
I'm updating the method names, sorry for the readability issue